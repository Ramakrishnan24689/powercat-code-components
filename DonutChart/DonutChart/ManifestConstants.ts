export const enum ManifestPropertyNames {
    dataset = 'dataset',
}

export const enum ItemColumns {
    Legend = 'ItemLegend',
    Key = 'ItemKey',
    Value = 'ItemValue',
    Color = 'ItemColor',
}
