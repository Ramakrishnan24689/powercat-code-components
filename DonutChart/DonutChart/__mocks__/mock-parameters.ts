/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        Title: new MockStringProperty(),
        HideLabels: new MockTwoOptionsProperty(),
        HideTooltip: new MockTwoOptionsProperty(),
        ShowLabelsInPercentage: new MockTwoOptionsProperty(),
        InnerRadius: new MockWholeNumberProperty(),
        ValueInsideDonut: new MockStringProperty(),
        Theme: new MockStringProperty(),
        CustomColors: new MockTwoOptionsProperty(),
        items: new MockDataSet([]),
        TabIndex: new MockWholeNumberProperty(),
        Tooltip: new MockStringProperty(),
    };
}
