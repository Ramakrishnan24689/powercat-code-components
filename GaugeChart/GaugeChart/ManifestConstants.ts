import { GaugeValueFormat } from '@fluentui/react-charting';

export const enum ManifestPropertyNames {
    dataset = 'dataset',
}

export const ChartValueFormats = {
    '1': GaugeValueFormat.Fraction,
    '0': GaugeValueFormat.Percentage,
};

export const enum ItemColumns {
    Legend = 'ItemLegend',
    Key = 'ItemKey',
    Count = 'ItemCount',
    Color = 'ItemColor',
    Size = 'ItemSize',
}
