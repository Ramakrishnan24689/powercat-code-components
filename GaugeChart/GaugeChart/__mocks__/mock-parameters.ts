/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockEnumProperty, MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        Title: new MockStringProperty(),
        Sublabel: new MockStringProperty(),
        ChartValue: new MockWholeNumberProperty(),
        ChartValueFormat: new MockEnumProperty(),
        MinValue: new MockWholeNumberProperty(),
        MaxValue: new MockWholeNumberProperty(),
        HideLegend: new MockTwoOptionsProperty(),
        HideMinMax: new MockTwoOptionsProperty(),
        Theme: new MockStringProperty(),
        CustomColors: new MockTwoOptionsProperty(),
        items: new MockDataSet([]),
        TabIndex: new MockWholeNumberProperty(),
        Tooltip: new MockStringProperty(),
    };
}
