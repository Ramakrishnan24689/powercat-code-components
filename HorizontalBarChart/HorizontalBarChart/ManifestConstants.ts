export const enum ManifestPropertyNames {
    dataset = 'dataset',
}

export const enum ItemColumns {
    Title = 'ItemTitle',
    Legend = 'ItemLegend',
    Value = 'ItemValue',
    TotalValue = 'ItemTotalValue',
    XPopover = 'ItemXPopOver',
    YPopover = 'ItemYPopover',
    Key = 'ItemKey',
    Count = 'ItemCount',
    Color = 'ItemColor',
}

export const Variants = {
    '0': 'absolute-scale',
    '1': 'part-to-whole',
};

export const ChartModes = {
    Default: 'default',
    Fraction: 'fraction',
    Percentage: 'percentage',
};
