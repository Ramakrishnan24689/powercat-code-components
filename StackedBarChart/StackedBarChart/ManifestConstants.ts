export const enum ManifestPropertyNames {
    dataset = 'dataset',
}

export const enum ItemColumns {
    Title = 'ItemTitle',
    Key = 'ItemKey',
    Value = 'ItemValue',
    Color = 'ItemColor',
    Callout = 'ItemCallout',
}

export const enum InputProperties {
    SelectedKey = 'SelectedKey',
}
