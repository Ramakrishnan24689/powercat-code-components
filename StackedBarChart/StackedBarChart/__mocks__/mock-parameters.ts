/* istanbul ignore file */

import { IInputs } from '../generated/ManifestTypes';
import { MockStringProperty, MockTwoOptionsProperty, MockWholeNumberProperty } from './mock-context';
import { MockDataSet } from './mock-datasets';

export function getMockParameters(): IInputs {
    return {
        Title: new MockStringProperty(),
        HideTooltip: new MockTwoOptionsProperty(),
        BarHeight: new MockWholeNumberProperty(),
        HideLegend: new MockTwoOptionsProperty(),
        BarBackgroundColor: new MockStringProperty(),
        AccessibilityLabel: new MockStringProperty(),
        Theme: new MockStringProperty(),
        CustomColors: new MockTwoOptionsProperty(),
        items: new MockDataSet([]),
        TabIndex: new MockWholeNumberProperty(),
        Tooltip: new MockStringProperty(),
    };
}
