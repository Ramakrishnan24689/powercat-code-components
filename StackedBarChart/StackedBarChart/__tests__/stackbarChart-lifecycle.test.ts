import { StackedBarChart } from '..';
import { getMockParameters } from '../__mocks__/mock-parameters';
import { MockContext } from '../__mocks__/mock-context';
import { IInputs } from '../generated/ManifestTypes';
import { MockDataSet, MockEntityRecord } from '../__mocks__/mock-datasets';
import { ItemColumns } from '../ManifestConstants';

// Since requestAnimationFrame does not exist in the test DOM, mock it
window.requestAnimationFrame = jest.fn().mockImplementation((callback) => {
    callback();
});

describe('Guage', () => {
    beforeEach(() => {
        jest.clearAllMocks();
        jest.useFakeTimers();
    });
    afterEach(() => {
        for (let i = 0; i < document.body.children.length; i++) {
            if (document.body.children[i].tagName === 'DIV') {
                document.body.removeChild(document.body.children[i]);
                i--;
            }
        }
        // reset any jest timers
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        if ((setTimeout as any).mock) {
            jest.runOnlyPendingTimers();
            jest.useRealTimers();
        }
    });

    it('renders', () => {
        const { component, context, notifyOutputChanged } = createComponent();
        component.init(context, notifyOutputChanged);
        const element = component.updateView(context);
        expect(element).toMatchSnapshot();
    });

    it('renders dummy items when no items configured', () => {
        const { component, context, notifyOutputChanged } = createComponent();
        // Simulate there being no items bound - which causes an error on the parameters
        context.parameters.items.error = true;
        component.init(context, notifyOutputChanged);
        const element = component.updateView(context);
        expect(element).toMatchSnapshot();
    });
});

function createComponent() {
    const component = new StackedBarChart();
    const notifyOutputChanged = jest.fn();
    const context = new MockContext<IInputs>(getMockParameters());
    context.parameters.items = new MockDataSet([
        new MockEntityRecord('1', {
            [ItemColumns.Key]: 'item1',
            [ItemColumns.Title]: 'Item 1',
            [ItemColumns.Value]: 30,
            [ItemColumns.Color]: 'blue',
        }),
        new MockEntityRecord('1', {
            [ItemColumns.Key]: 'item1',
            [ItemColumns.Title]: 'Item 1',
            [ItemColumns.Value]: 20,
            [ItemColumns.Color]: 'Red',
        }),
    ]);

    return { component, context, notifyOutputChanged };
}
