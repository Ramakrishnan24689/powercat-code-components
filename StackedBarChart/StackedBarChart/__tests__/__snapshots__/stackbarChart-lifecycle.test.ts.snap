// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Guage renders 1`] = `
<Memo(StackedBarChartControl)
  data={
    Object {
      "chartData": Array [
        Object {
          "color": "#e3008c",
          "data": 20,
          "legend": "Item 1",
          "xAxisCalloutData": undefined,
        },
        Object {
          "color": "#13a10e",
          "data": 20,
          "legend": "Item 1",
          "xAxisCalloutData": undefined,
        },
      ],
      "chartDataAccessibilityData": Object {
        "ariaLabel": "",
      },
      "chartTitle": "",
    }
  }
  styles={
    Object {
      "root": Object {
        "width": -1,
      },
    }
  }
  themeJSON=""
  width={-1}
/>
`;

exports[`Guage renders dummy items when no items configured 1`] = `
<Memo(StackedBarChartControl)
  data={
    Object {
      "chartData": Array [
        Object {
          "color": "#57811b",
          "data": 100,
          "legend": "Item 1",
          "xAxisCalloutData": "1",
        },
        Object {
          "color": "#026467",
          "data": 200,
          "legend": "Item 2",
          "xAxisCalloutData": "2",
        },
        Object {
          "color": "#9a3d0c",
          "data": 300,
          "legend": "Item 3",
          "xAxisCalloutData": "3",
        },
      ],
      "chartTitle": "StackChart",
    }
  }
  styles={
    Object {
      "root": Object {
        "width": -1,
      },
    }
  }
  themeJSON=""
  width={-1}
/>
`;
